src/main/java/org/sqlite/jdbc4/JDBC4ResultSet.java

```diff
@@ -320,6 +325,36 @@ public <T> T getObject(int columnIndex, Class<T> type) throws SQLException {
         if (type == Time.class) return type.cast(getTime(columnIndex));
         if (type == Timestamp.class) return type.cast(getTimestamp(columnIndex));
 
+        if (type == LocalDate.class) {
+            try {
+
+                return type.cast(getDate(columnIndex).toLocalDate());
+            } catch (SQLException sqlException) {
+                // If the FastDateParser failed, try parse it with LocalDate.
+                // It's a workaround for a value like '2022-12-1' (i.e no time presents).
+
+                return type.cast(LocalDate.parse(getString(columnIndex)));
```

Comment:
do we still need this?

Reply:
Yes, we need this one and the one below. 

Assuming that change the LocalDate logic to this: 
![f13b7b3d1ea9633d8d06c768deb9a78](https://user-images.githubusercontent.com/4351489/234835643-2d148460-5374-4dbd-bf58-60f8302f532f.png)

This test will fail, 
```java
@Test
void test1() throws SQLException {
    stat.executeUpdate("create table datetime_test(c1)");
    stat.executeUpdate("insert into datetime_test values ('2021-11-09')");

    ResultSet rs = stat.executeQuery("select * from datetime_test");

    rs.next();
    assertThat(rs.getObject(1, LocalDate.class)).isEqualTo(LocalDate.of(2021, 11, 9));
}
```
and produce this: 
```
java.sql.SQLException: Error parsing date

	at org.sqlite.jdbc3.JDBC3ResultSet.getDate(JDBC3ResultSet.java:250)
	at org.sqlite.jdbc4.JDBC4ResultSet.getObject(JDBC4ResultSet.java:329)
	at org.sqlite.ResultSetTest.test1(ResultSetTest.java:362)
	(omitted due to the long trace)
Caused by: java.text.ParseException: Unparseable date: "2021-11-09" does not match (\p{Nd}++)\Q-\E(\p{Nd}++)\Q-\E(\p{Nd}++)\Q \E(\p{Nd}++)\Q:\E(\p{Nd}++)\Q:\E(\p{Nd}++)\Q.\E(\p{Nd}++)
	at org.sqlite.date.FastDateParser.parse(FastDateParser.java:311)
	at org.sqlite.date.FastDateFormat.parse(FastDateFormat.java:449)
	at org.sqlite.jdbc3.JDBC3ResultSet.getDate(JDBC3ResultSet.java:248)
	... 70 more
```